### PR TITLE
updating guardian cdk to 46.0.1 and its dependencies to relevant versions

### PR DESCRIPTION
**Pull request by michaelwmcnamara**
This updates cdk and related libraries to latest version - 46.0.1.

## How to test

build and deploy amiable.

## How can we measure success?

Amiable cdk is up to date

### CODE REVIEW

File: cdk/package.json [hunk 1]
```diff
@@ -27,13 +27,13 @@
     "prettier": "^2.6.2",
     "ts-jest": "^27.1.4",
     "ts-node": "^10.2.0",
-    "typescript": "~4.4.3"
+    "typescript": "~4.4.3",
+    "@guardian/cdk": "46.0.1",
```
Comment (jfsoul) [inline cdk/package.json:31]: bit surprised this didn't require an update to the snapshots for the tests!

File: cdk/package.json [hunk 2]
```diff
@@ -27,13 +27,13 @@
...
+    "aws-cdk": "2.29.1",
+    "aws-cdk-lib": "2.29.1",
+    "constructs": "10.1.42"
   },
   "dependencies": {
-    "aws-cdk": "2.23.0",
-    "aws-cdk-lib": "2.23.0",
-    "constructs": "10.1.5",
-    "@guardian/cdk": "43.2.0",
     "source-map-support": "^0.5.16"
```
Comment (jfsoul) [inline cdk/package.json:37]: same question as the other PR, could this dependency be moved too?

**Review by jfsoul — APPROVED** [review]
Same question as the other PR, but otherwise

**Comment by jorgeazevedo** [comment]
@michaelwmcnamara this PR broke snyk with error

```
Monitoring /home/runner/work/amiable/amiable/cdk/yarn.lock...

Dependency @babel/highlight@^7.14.5 was not found in yarn.lock.
Your package.json and yarn.lock are probably out of sync.
Please run "yarn install" and try again.
```

See https://github.com/guardian/amiable/runs/7403360668?check_suite_focus=true

**Comment by jorgeazevedo** [comment]
@michaelwmcnamara it appears jon soul has just opened a PR to fix the issue, see #161

**Pull request merged**